### PR TITLE
Add flow network (task 5)

### PR DESCRIPTION
**Pull request by Vectrom**

### CODE REVIEW

File: app/exercises/base.py [hunk 1]
```diff
@@ -53,4 +53,7 @@ def clear_text(self):
 
     def clear_graph(self):
         self.graph = None
-        self.canvas.delete('all')
\ No newline at end of file
+        self.canvas.delete('all')
+
+    def add_text_to_result(self, text):
```
Comment (kmalski) [inline app/exercises/base.py:58]: There are no consequences here clear_graph, draw_graph, print_graph, add_text_to_result. Names are more than misleading. Maybe change to something like that:
- draw_graph (no changes)
- clear_graph (no changes)
- print_text
- clear_text
- append_text
Or give other suggestions

File: app/exercises/five.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+from exercises.base import BaseTab
+from structures.weighted_adjacency_list import WeightedDirectedAdjacencyList
+
+import tkinter as tk
+from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg, NavigationToolbar2Tk
+from matplotlib.figure import Figure
+import networkx as nx
+from tkinter import ttk, messagebox
+
```
Comment (kmalski) [inline app/exercises/five.py:9]: Keep order
import ...
import ...
from ...
from ...

File: app/exercises/five.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            # only display the buttons we need
+            toolitems = [t for t in NavigationToolbar2Tk.toolitems if
+                 t[0] in ('Home', 'Pan', 'Zoom', 'Save')]
+
+        frame = ttk.Frame(self)
+        frame.grid(row=row, column=column, sticky='NSWE')
+        frame.grid_propagate(False)
+
+        figure = Figure(figsize=(5, 4), constrained_layout=True, frameon=False)
+        self.ax = figure.add_subplot(111)
```
Comment (kmalski) [inline app/exercises/five.py:60]: What's 'ax' ?

File: app/structures/weighted_adjacency_list.py [hunk 4]
```diff
@@ -161,6 +170,15 @@ class WeightedDirectedAdjacencyList(WeightedAdjacencyList):
     def __init__(self, graph):
         self.graph = graph
 
+    def is_output_vertex(self, vertex, listToCheck = None):
```
Comment (kmalski) [inline app/structures/weighted_adjacency_list.py:173]: Why do you suddenly use camelCase variables names?

File: app/structures/weighted_adjacency_list.py [hunk 5]
```diff
@@ -276,4 +294,48 @@ def calculate_distance_matrix(self) -> np.ndarray:
                     dist_matrix[vertex_1, vertex_2] = dijkstra_distance[vertex_2] - h[vertex_1] + h[vertex_2]
             
             return dist_matrix
+
+    def generate_flow_network(self, n: int) -> dict:
+        layers = {0: [0]}
+        self.add_vertex(0)
+
+        lastIndex = 0
```
Comment (kmalski) [inline app/structures/weighted_adjacency_list.py:302]: Again camelCase

File: app/structures/weighted_adjacency_list.py (hunk 4 above)
Comment (kmalski) [inline app/structures/weighted_adjacency_list.py:173]: Add types for non obvious parameters. Is it List[int]?

File: app/structures/weighted_adjacency_list.py [hunk 6]
```diff
@@ -276,4 +294,48 @@ def calculate_distance_matrix(self) -> np.ndarray:
...
+            end = random.choice(vertices)
+            if start != end and not self.is_edge(start, end) and not self.is_edge(end, start) and end != 0 and start != len(vertices) - 1:
+                self.add_edge(start, end, 0)
+                additional_edge_number += 1
+
+        self.set_random_weights(1, 10)
+
+        return layers
+
+    def convert_to_networkX(self):
```
Comment (kmalski) [inline app/structures/weighted_adjacency_list.py:333]: Whats the return type? Again not obvious

File: app/utils/tkinter.py [hunk 7]
```diff
@@ -118,6 +118,9 @@ def show(self):
     def clear(self):
         self['text'] = ''
 
+    def add_text(self, text):
```
Comment (kmalski) [inline app/utils/tkinter.py:121]: Rather append and 'text' is redundant here, be consistent

File: app/exercises/five.py (hunk 2 above)
Comment (Vectrom) [reply app/exercises/five.py:9]: imports are in alphabetical order

File: app/exercises/five.py (hunk 3 above)
Comment (Vectrom) [reply app/exercises/five.py:60]: axis

Comment (Vectrom) [reply app/exercises/five.py:60]: networkX also use "ax" as axis

File: app/structures/weighted_adjacency_list.py (hunk 4 above)
Comment (Vectrom) [reply app/structures/weighted_adjacency_list.py:173]: It's force of habit. ;)

File: app/exercises/base.py (hunk 1 above)
Comment (Vectrom) [reply app/exercises/base.py:58]: print_graph is good (it prints graph, not text from parameter)

File: app/exercises/five.py (hunk 2 above)
Comment (kmalski) [reply app/exercises/five.py:9]: But should not

File: app/exercises/five.py (hunk 3 above)
Comment (Vectrom) [reply app/exercises/five.py:60]: `nx.draw_networkx(self.visualization, pos=pos, ax=self.ax)`

Comment (kmalski) [reply app/exercises/five.py:60]: But there is no context and u can't guess it from 60th line.

File: app/exercises/base.py (hunk 1 above)
Comment (Vectrom) [reply app/exercises/base.py:58]: append_text as method in BaseTab class says nothing

Comment (kmalski) [reply app/exercises/base.py:58]: But it doesnt print only graph.  It print also paths and results. U can change "text" to something better

File: app/structures/weighted_adjacency_list.py (hunk 4 above)
Comment (Vectrom) [reply app/structures/weighted_adjacency_list.py:173]: it's bool ;)

File: app/exercises/base.py (hunk 1 above)
Comment (Vectrom) [reply app/exercises/base.py:58]: No, it prints only graph (list)

**Pull request merged**